dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs

```diff
@@ -0,0 +1,170 @@
+
+        await ChatAsync(
+            "Assistants.ParrotAssistant.yaml",
+            "Fortune favors the bold.",
+            "I came, I saw, I conquered.",
+            "Practice makes perfect.");
+    }
+
+    private static async Task RunWithNativeFunctionsAsync()
+    {
+        Console.WriteLine("======== Run:WithNativeFunctions ========");
+
+        IKernel bootstraper = new KernelBuilder().Build();
+
+        var functions = bootstraper.ImportFunctions(new MenuPlugin(), nameof(MenuPlugin));
+
+        await ChatAsync(
+            "Assistants.ToolAssistant.yaml",
+            functions.Values,
+            "Hello",
+            "What is the special soup?",
+            "What is the special drink?",
+            "Thank you!");
+    }
+
+    private static async Task RunWithSemanticFunctionsAsync()
+    {
+        Console.WriteLine("======== Run:WithSemanticFunctions ========");
+
+        IKernel bootstraper = new KernelBuilder().Build();
+
+        var functions =
+            new[]
+            {
+                bootstraper.CreateSemanticFunction(
+                    "Correct any misspelling or gramatical errors provided in input: {{$input}}",
+                    functionName: "spellChecker",
+                    pluginName: "test",
+                    description: "Correct the spelling for the user input."),
+            };
+
+        await ChatAsync(
+            "Assistants.ToolAssistant.yaml",
+            functions,
+            "Hello",
+            "Is this spelled correctly: exercize",
+            "What is the special soup?",
+            "Thank you!");
+    }
+
+    private static async Task RunAsFunctionAsync()
+    {
+        Console.WriteLine("======== Run:AsFunction ========");
+
+        var assistant =
+            await AssistantBuilder.FromDefinitionAsync(
+                TestConfiguration.OpenAI.ApiKey,
+                OpenAIFunctionEnabledModel,
+                EmbeddedResource.Read("Assistants.ParrotAssistant.yaml"));
+
+        IKernel bootstraper = new KernelBuilder().Build();
+        var assistants = bootstraper.ImportFunctions(assistant, assistant.Id);
+
+        var variables = new ContextVariables
+        {
+            ["input"] = "Practice makes perfect."
+        };
+        var result = await bootstraper.RunAsync(assistants.Single().Value, variables);
+        var resultValue = result.GetValue<string>();
+
+        var response = JsonSerializer.Deserialize<AssistantResponse>(resultValue ?? string.Empty);
+        Console.WriteLine(
+            response?.Response ??
+            $"No response from assistant: {assistant.Id}");
+    }
+
+    private static Task ChatAsync(
+         string resourcePath,
+         params string[] messages)
+    {
+        return ChatAsync(resourcePath, null, messages);
+    }
+
+    private static async Task ChatAsync(
+        string resourcePath,
+        IEnumerable<ISKFunction>? functions,
+        params string[] messages)
```

Comment:
Can you change order of parameters make functions default to null and then remove the override

Reply:
Doesn't params need to be the last parameter?

Reply:
Discussed in teams